packages/remix-dev/config/defaults/entry.client.tsx

```diff
@@ -13,10 +13,10 @@ function hydrate() {
   });
 }
 
-if (typeof requestIdleCallback === "function") {
-  requestIdleCallback(hydrate);
+if (window.requestIdleCallback) {
+  window.requestIdleCallback(hydrate);
 } else {
   // Safari doesn't support requestIdleCallback
   // https://caniuse.com/requestidlecallback
-  setTimeout(hydrate, 1);
+  window.setTimeout(hydrate, 1);
 }
```

Comment:
I think we should keep these to check if global `requestIdleCallback` is a function (see #4998)

```suggestion
if (typeof requestIdleCallback === "function") {
  requestIdleCallback(hydrate);
} else {
  // Safari doesn't support requestIdleCallback
  // https://caniuse.com/requestidlecallback
  setTimeout(hydrate, 1);
}
```